### ISSUE TITLE
Option to enabling setting unread meessages as read when opening them

### ISSUE THREAD

**Issue by je-vv** [LemmyNet/lemmy#1286]
When 1st interacting with lemmy's inbox, it was really hard for me to realize that I needed to manually set each message as read, or to set them all as read through the all messages button, since I was under the supposition when I opened an unread message, it would automatically turn into a read message, but that never happened.

To me it's more intuitive if the messages were turned into read, once opening them, and if there are several comments pertaining to the same original conversation or post, then by opening that original post, all of the comments would turn into read at once.

Is it possible to get an option, one could turn on, in order to get the described behaviour?  Thanks !

**Comment by dessalines** [comment]
> when I opened an unread message, it would automatically turn into a read message, but that never happened.

If you click the comment link button from your inbox, it should mark the comment as read. There's also the `mark all comments as read button`.

**Comment by je-vv** [comment]
> > when I opened an unread message, it would automatically turn into a read message, but that never happened.
> 
> If you click the comment link button from your inbox, it should mark the comment as read. There's also the `mark all comments as read button`.

True, I'm using that now, however I found it a bit unintuitive, specially for people used to reddit.  If there was an option to do it automatically when opening messages or the original post, or the full comments, that would be just awesome.  Thanks !

**Comment by dessalines** [comment]
How are you opening messages?

Opening the original post wouldn't be good because you might have multiple unread messages in one post.

**Comment by weex** [comment]
It would be good to have an option in settings to mark read when viewing notifications. Perhaps I've been habituated by Reddit to take each view of comment replies as a one time deal, but I hate seeing notifications for things I've already seen. Just now I had the number 4 next to the bell when only one was new, so that adds an extra minor level of confusion.

**Comment by dessalines** [comment]
I'm not sold on this one, because I really hate reddits default behavior. Usually I'll go to my inbox, know I need to respond to some messages, navigate away, come back, and whoops, every message got marked as read simply because I viewed my inbox page. If I don't have time, then I have to explicitly go back and mark all the ones that need attention as unread.

Lemmy currently has 3 ways to mark them as read: clicking the explicit mark as read checkmark, clicking the link, or clicking the mark all as read button. That's a lot more explicit an action than just looking at your inbox page and assuming the user wants every message there to be marked as read.

**Comment by weex** [comment]
I appreciate the principled approach which applies to default behavior and the three ways should be sufficient, but I also feel for users (me) who got used to a certain way of dealing with comments on a reddit-like site and now have to keep two ways on the mental stack. That's why I advocate for a setting. I'd set it to be like reddit and never think about it again. Meanwhile, you might have lemmy-first users later asking reddit or RES for a setting which would be another kind of win.

**Comment by weex** [comment]
I put a $50 bounty on this in case it helps. I'm not sure if the code has to be merged to claim from bountysource but it satisfies me at least if a pull request is made.
